server/src/main/java/io/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -297,7 +297,9 @@ boolean hasLoadPending(final String dataSource)
       for (DruidServer druidServer : serverInventoryView.getInventory()) {
         final DruidDataSource loadedView = druidServer.getDataSource(dataSource.getName());
         if (loadedView != null) {
-          segments.removeAll(loadedView.getSegments());
+          for (DataSegment serverSegment : loadedView.getSegments()) {
```

Comment:
This deserves a comment so someone doesn't "simplify" it back into the old code.

Reply:
Good point, added a comment